rust/rope/src/rope.rs

```diff
@@ -145,6 +147,51 @@ impl Metric<RopeInfo> for BaseMetric {
     }
 }
 
+#[derive(Clone, Copy)]
+pub struct GraphemeMetric(());
+
+impl Metric<RopeInfo> for GraphemeMetric {
+    fn measure(_: &RopeInfo, len: usize) -> usize {
+        len
+    }
+
+    fn to_base_units(_: &String, in_measured_units: usize) -> usize {
+        in_measured_units
+    }
+
+    fn from_base_units(_: &String, in_base_units: usize) -> usize {
+        in_base_units
+    }
+
+    fn is_boundary(s: &String, offset: usize) -> bool {
+        if let Ok(r) = GraphemeCursor::new(offset, s.len(), true).is_boundary(s, 0) {
```

Comment:
you can call `.ok()` on any `Result<T>` to convert it to an `Option<T>`, discarding the error.